ReactiveXComponent/WebSocket/WebSocketXCApiManager.cs

```diff
@@ -0,0 +1,113 @@
+                handler => (sender, e) => handler(e),
+                h => XCApiReceived += h,
+                h => XCApiReceived -= h);
+        }
+
+        public List<string> GetXCApiList(TimeSpan? timeout = null)
+        {
+            var delay = timeout ?? TimeSpan.FromSeconds(10);
+            List<string> result = null;
+            var lockEvent = new AutoResetEvent(false);
+            var observer = Observer.Create<List<string>>(message =>
+            {
+                result = new List<string>(message);
+                lockEvent.Set();
+            });
+
+            using (_xcApiListStream.Subscribe(observer))
+            {
+                _webSocketClient.MessageReceived += ProcessResponse;
+                SendRequest(WebSocketCommand.GetXCApiList);
+                lockEvent.WaitOne(delay);
+                _webSocketClient.MessageReceived -= ProcessResponse;
+            }
+
+            return result;
+        }
+
+
+        public string GetXCApi(string apiFullName, TimeSpan? timeout = null)
+        {
+            var delay = timeout ?? TimeSpan.FromSeconds(10);
+            var result = string.Empty;
+            var lockEvent = new AutoResetEvent(false);
+            var observer = Observer.Create<string>(message =>
+            {
+                result = message;
+                lockEvent.Set();
+            });
+
+            using (_xcApiStream.Subscribe(observer))
+            {
+                _webSocketClient.MessageReceived += ProcessResponse;
+                SendRequest(WebSocketCommand.GetXCApi, apiFullName);
+                lockEvent.WaitOne(delay);
+                _webSocketClient.MessageReceived -= ProcessResponse;
+            }
+
+            return result;
+        }
+
+        private void SendRequest(string requestCommand, string apiFullName = null)
+        {
+            var webSocketRequest = WebSocketMessageHelper.SerializeXCApiRequest(requestCommand, apiFullName);
+            _webSocketClient.Send(webSocketRequest);
+        }
+
+        private void ProcessResponse(object sender, WebSocketMessageEventArgs args)
+        {
+            var rawRequest = args.Data;
+            var webSocketMessage = WebSocketMessageHelper.DeserializeRequest(rawRequest);
+
+            if (webSocketMessage.Command == WebSocketCommand.GetXCApiList)
+            {
+                var response = JsonConvert.DeserializeObject<WebSocketGetXcApiListResponse>(webSocketMessage.Json);
+                var xcApiList = new List<string>();
+                if (response?.Apis != null)
+                {
+                    xcApiList = new List<string>(response.Apis);
+                }
+                XCApiListReceived?.Invoke(this, xcApiList);
+            }
+            else if (webSocketMessage.Command == WebSocketCommand.GetXCApi)
+            {
+                var response = JsonConvert.DeserializeObject<WebSocketGetXcApiResponse>(webSocketMessage.Json);
```

Comment:
Just move this in **WebSocketMessageHelper** in a method like _DeserializeXcApiResponse()_.